### ISSUE TITLE
Fix running benchmark

### ISSUE THREAD

**Issue by mohits** [klaxit/fast-polylines#21]
I find that on Windows (at least), if you try to do
$ make benchmark
it crashes out with this:

$ bundle exec ruby -I lib:ext -r fast_polylines ./perf/benchmark.rb
Traceback (most recent call last):
        3: from ./perf/benchmark.rb:3:in `<main>'
        2: from ./perf/benchmark.rb:3:in `require'
        1: from D:/qxp/bbt_proj/fast-polylines/lib/fast_polylines.rb:3:in `<top (required)>'
D:/proj/fast-polylines/lib/fast_polylines.rb:3:in `require': cannot load such file -- fast_polylines/fast_polylines (LoadError)

From what I see, the Makefile has:
In the Makefile, I see this:
> RUBY_FLAG = -I lib:ext -r $(EXT_NAME)
> benchmark: ext
>	bundle exec ruby $(RUBY_FLAG) ./perf/benchmark.rb

This does not have the lib directory on it and so (it appears that) it fails with the above error.

The fix is simple:
> RUBY_FLAG = -I lib:ext -I lib -r $(EXT_NAME)

I'm keen to know if this is a problem that you also see or is it just me?

Thanks!

**Comment by BuonOmo** [comment]
Hi @mohits,

According to ruby's man page:

> -I directory   Used to tell Ruby where to load the library scripts.  Directory path will be added to the load-path variable ($:).

Hence `lib:ext` is already adding both to the path.

Maybe the ruby CLI has a different behavior on Windows? I guess you could try the next bash command:

```bash
$ ruby -I lib:ext <<<"puts $:"
/Users/me/Dev/fast-polylines/lib
/Users/me/Dev/fast-polylines/ext
...
```

**Comment by mohits-qi** [comment]
@BuonOmo - It seems to be!

C:\Users\mohit>ruby -I lib:ext -e "puts $:"
C:/Users/mohit/lib:ext
...

Digging into this, I found this from Programming Ruby:
-I directories
' Specifies directories to be prepended to $LOAD_PATH ($:). Multiple -I options may be present, and multiple directories may appear following each -I. **Directories are separated by a ":'' on Unix-like systems and by a ";'' on DOS/Windows systems**. 

Of course, when we do the right thing, it works :)
C:\Users\mohit>ruby -I lib;ext  -e "puts $:"
C:/Users/mohit/lib
C:/Users/mohit/ext
C:/Ruby22-x64/lib/ruby/site_ruby/2.2.0

**Comment by BuonOmo** [comment]
Hi @mohits, it's been a while since I wanted to improve the makefile. It is done now, could you tell me if it is now working on windows ? :) (https://github.com/klaxit/fast-polylines/pull/23)

**Comment by mohits** [comment]
Thank you for doing this. Merging #19  and #23 makes Windows a "first class citizen", i.e., it works out of the box although it won't be tested automatically due to Travis limitations. If you tag me on a pre-release, I'd be happy to help run it locally on my Windows Ruby to make sure that nothing is broken.

**Comment by BuonOmo** [comment]
@mohits I'm thinking about adding a note about that on the readme. I'll do so once everything is merged. No need for a manual test on pre-release, I'll do a patch and include the fact that windows compilation is not guaranteed since we cannot test it in an automated fashion. Could I add your name in that paragraph to say that you should be mentioned in windows related issue ?

In the meantime, I'll wait for the last fixes to integrate every PRs you made ! Thanks again for the journey you made :)

**Comment by mohits** [comment]
You're most welcome. I'm sorry I had overlooked the note on the fixes you wanted from me when I cleared out everything on Sunday. Have made the minor changes on #19 and once it goes through Travis, hopefully, everything can be merged and we are good to publish the latest version :) that will work with Windows out of the box.

Thanks for being receptive to the changes.

**Comment by mohits** [comment]
Hi @BuonOmo - I wanted to write a blog post that examines how a native gem works and wanted to use fast-polylines as the gem under the microscope. The first part is at: https://notepad.onghu.com/2023/learning-by-reversing-s1-e1-native-gems/ but when putting this together, I realised that running `make benchmark` actually still does not work. It works if you run the benchmark script directly from the windows command line but not from the Makefile.

The reason for this is that the Makefile specifies that `SHELL=/bin/sh` which does not respect the Windows style `-Ilib;ext` and on the other hand, the rest of the Windows stuff needs it.

The correct fix is to use `-Ilib -Iext` so that both paths are passed to it. I will raise a PR for this again but let me know if I should open this as a separate issue or leave it on this one since the title of the issue would likely be similar.

**Comment by BuonOmo** [comment]
Nice article! I don't have access to my computer these days, but I'll test your PR on Mac as soon as I have my laptop. No need to open a new issue, the PR will be just fine.

**Comment by mohits** [comment]
I will create the PR in a few days (probably next weekend) and there's no rush to resolve it :)
